generator/src/main/java/org/corfudb/generator/operations/CheckpointOperation.java

```diff
@@ -28,10 +29,10 @@ public void execute() {
 
             MultiCheckpointWriter mcw = new MultiCheckpointWriter();
             mcw.addAllMaps(state.getMaps());
-            long trimAddress = mcw.appendCheckpoints(state.getRuntime(), "Maithem");
-            state.setTrimMark(trimAddress);
+            Token trimAddress = mcw.appendCheckpoints(state.getRuntime(), "Maithem");
+            state.setTrimMark(trimAddress.getSequence());
             Thread.sleep(1000l * 30l * 1l);
-            state.getRuntime().getAddressSpaceView().prefixTrim(trimAddress - 1);
+            state.getRuntime().getAddressSpaceView().prefixTrim(trimAddress.getSequence() - 1);
```

Comment:
same applies here, I'm not sure if we should propagate the Token semantics through all layers, as a sequence number on its own kind of loses meaning with the new redefinition.

Reply:
It is cleaner and we should, but this patch only adds the epoch to the timestamp. The consumption of this epoch everywhere will be a multistage effort, every reference to getSequence() should be replaced with just a token. 

I need to change the LU logic to make this change, but its not for this patch. This patch just fixes the transaction resolution issue.

Reply:
Ok don't worry I'll do this in my patch!

Reply:
Ok, we can also distribute that translation. I was imagining having multiple patches where the translation happens per layer on the client, then per server etc.. i think it will be cleaner than having one big patch.